app/models/commit/import.rb

```diff
@@ -8,57 +8,69 @@
 module Pod
   module TrunkApp
     class Commit
-      module Import
+      class Import
         DATA_URL_TEMPLATE = "https://raw.githubusercontent.com/#{ENV['GH_REPO']}/%s/%s"
 
-        def self.log_failed_spec_fetch(url, message, data)
-          LogMessage.create(
-            :message => "There was an issue fetching the spec at #{url}: #{message}",
-            :level => :error,
-            :data => data
-          )
-        end
+        attr_reader :committer_email
+        attr_reader :committer_name
 
-        # TODO: handle network/request failures
-        #
-        def self.fetch_spec(commit_sha, file)
-          url = DATA_URL_TEMPLATE % [commit_sha, file]
-          response = REST.get(url)
-          data = response.body
-          if response.ok?
-            ::Pod::Specification.from_string(data, file)
-          else
-            log_failed_spec_fetch(url, response.status_code.to_s, data)
-            nil
-          end
-        rescue REST::Error => e
-          log_failed_spec_fetch(url, "#{e.class.name} - #{e.message}", e.backtrace.join("\n\t\t"))
-          nil
+        def initialize(committer_email, committer_name)
+          @committer_email = committer_email
+          @committer_name  = committer_name
         end
 
         # For each changed file, get its data (if it's a podspec).
         #
-        def self.import(commit_sha, type, files, committer_email, committer_name)
+        def import(commit_sha, type, files)
           files.each do |file|
             next unless file =~ /\.podspec(.json)?\z/
 
-            spec = fetch_spec(commit_sha, file)
-            next unless spec
+            pod_name, version_name = extract_name_and_version file
 
-            unless committer = Owner.find_by_email(committer_email)
-              committer = Owner.create(:email => committer_email, :name => committer_name)
-            end
+            pod = Pod.find_or_create(:name => pod_name)
+
+            committer = find_or_create_committer
+
+            if type == :removed
+              handle_removed(pod, version_name, committer, commit_sha)
+            else
+              spec = fetch_spec(commit_sha, file)
+
+              next unless spec
 
-            pod = Pod.find_or_create(:name => spec.name)
-            pod.add_owner(committer) if pod.was_created?
+              pod.add_owner(committer) if pod.was_created?
 
-            send(:"handle_#{type}", spec, pod, committer, commit_sha)
+              handle_with_existing_spec(type, spec, pod, committer, commit_sha)
+            end
           end
         end
 
+        # Extracts the pod name and the version name from the file name.
+        #
+        def extract_name_and_version(file_name)
+          _, name, version_name, _ = *file_name
+            .match(%r{([^\/]+)\/([^\/]+)\/[^\.]+\.podspec(.json)?\z})
```

Comment:
What I meant is (it was late): we check for the podfile by asserting against the extensions. We can use the results for this assertion further down. I also ran into trouble with the tests.

I'd say feel free to change this back! :)